File: solutions/338/338.py
class Solution:
    def countBits(self, n):
        ans = []
        
        for i in range(n + 1):
            number = i
            sum = 0

            while number != 0:
                sum += number % 2
                number = number // 2
            
            ans.append(sum)

        return ans

File: solutions/338/338.java
class Solution {
    public int[] countBits(int n) {
        int[] ans = new int[n + 1];
        
        for (int i = 0; i <= n; i++) {
            int number = i;
            int sum = 0;

            while (number != 0) {
                sum += number % 2;
                number = number / 2;
            }
            
            ans[i] = sum;
        }

        return ans;
    }
}